janus_core/calculations/phonons.py

```diff
@@ -534,17 +584,44 @@ def write_bands(
         if save_plots is None:
             save_plots = self.plot_to_file
 
-        bands_file = self._build_filename("auto_bands.yml", filename=bands_file)
-        self.results["phonon"].auto_band_structure(
-            write_yaml=True,
-            filename=bands_file,
+        if self.paths:
+            bands_file = self._build_filename("bands.yml.xz", filename=bands_file)
+
+            with open(self.paths, encoding="utf8") as paths_file:
+                paths_info = safe_load(paths_file)
+
+            labels = paths_info["labels"]
+            num_q_points = sum([len(q) for q in paths_info["paths"]])
+            num_labels = len(labels)
+            assert (
```

Comment:
Small thing of note is that `assert` is disabled if run with `python -O`, while not very common, it's liable to become more common as python has recently implemented bytecode optimisation. Do you want this to be disabled?

Reply:
I think it would just end up throwing a less helpful error, so I think that's fine if someone wants it to be optimised